crates/matrix-sdk/src/sliding_sync.rs

```diff
@@ -1140,30 +1191,44 @@ impl<'a> SlidingSyncViewRequestGenerator<'a> {
         (end, self.make_request_for_ranges(ranges))
     }
 
-    fn make_request_for_ranges(&self, ranges: Vec<(UInt, UInt)>) -> v4::SyncRequestList {
+    fn make_request_for_ranges(
+        &self,
+        ranges: Vec<(UInt, UInt)>,
+    ) -> (v4::SyncRequestList, Vec<(usize, usize)>) {
         let sort = self.view.sort.clone();
         let required_state = self.view.required_state.clone();
         let timeline_limit = self.view.timeline_limit;
         let filters = self.view.filters.clone();
 
-        assign!(v4::SyncRequestList::default(), {
-            ranges,
-            required_state,
-            sort,
-            timeline_limit,
-            filters,
-        })
+        (
+            assign!(v4::SyncRequestList::default(), {
+                ranges: ranges.clone(),
+                required_state,
+                sort,
+                timeline_limit,
+                filters,
+            }),
+            ranges
+                .into_iter()
+                .map(|(a, b)| {
+                    (
+                        usize::try_from(a).expect("range is a valid u32"),
+                        usize::try_from(b).expect("range is a valid u32"),
```

Comment:
Maybe just use `as` here? Will clippy complain then?

Reply:
doesn't compile, `a` and `b` are `UInt` and thus `as usize` isn't supported.